Amplify/Categories/Storage/Operation/Request/StorageDownloadDataRequest.swift

```diff
@@ -7,36 +7,68 @@
 
 import Foundation
 
+/// - Tag: StorageDownloadDataRequest
 public struct StorageDownloadDataRequest: AmplifyOperationRequest {
+
     /// The unique identifier for the object in storage
+    ///
+    /// - Tag: StorageDownloadDataRequest.key
     public let key: String
 
     /// Options to adjust the behavior of this request, including plugin-options
+    ///
+    /// - Tag: StorageDownloadDataRequest.options
     public let options: Options
 
+    /// - Tag: StorageDownloadDataRequest.init
     public init(key: String, options: Options) {
         self.key = key
         self.options = options
     }
 }
 
 public extension StorageDownloadDataRequest {
+
     /// Options to adjust the behavior of this request, including plugin-options
+    ///
+    /// - Tag: StorageDownloadDataRequestOptions
     struct Options {
+
         /// Access level of the storage system. Defaults to `public`
+        ///
+        /// - Tag: StorageDownloadDataRequestOptions.accessLevel
         public let accessLevel: StorageAccessLevel
 
-        /// Target user to apply the action on.
+        /// Target user on whose behalf the download will be performed.
+        ///
+        /// - Tag: StorageDownloadDataRequestOptions.targetIdentityId
         public let targetIdentityId: String?
-
-        /// Extra plugin specific options, only used in special circumstances when the existing options do not provide
-        /// a way to utilize the underlying storage system's functionality. See plugin documentation for expected
-        /// key/values
+        
+        /// Extra plugin specific options, only used in special circumstances when the existing options do
+        /// not provide a way to utilize the underlying storage system's functionality. See plugin
+        /// documentation for expected key/values.
+        ///
+        /// As an example, if using the AWSS3StoragePlugin, one may be want to add something like the
+        /// following (please note that `useAccelerateEndpoint`
+        /// [should first be setup](https://docs.amplify.aws/lib/storage/transfer-acceleration/q/platform/js/),
+        /// otherwise, requests will fail):
+        ///
+        /// ```
+        ///  let options = StorageDownloadDataRequest.Options(
+        ///      pluginOptions: [
+        ///          "useAccelerateEndpoint": true
+        ///      ]
+        ///  )
+        /// ```
+        ///
+        /// # Reference
+        /// * [Storage - Use Transfer Acceleration](https://docs.amplify.aws/lib/storage/transfer-acceleration/q/platform/js/)
```

Comment:
Should we also outline how to do this in the Swift platform docs so we're not sending iOS devs to the JS Storage page?

Reply:
Yes!

Reply:
OK, I reverted all these swiftdoc changes and filed the following ticket for when the documentation is ready: https://github.com/aws-amplify/amplify-swift/issues/2640